src/adc.rs

```diff
@@ -149,6 +151,42 @@ macro_rules! adc_pins {
     };
 }
 
+macro_rules! adc_internal {
+    ($($input:ty => ($chan:expr, $en:ident)),+ $(,)*) => {
+        $(
+            impl $input {
+                pub fn new() -> Self {
+                    Self {}
+                }
+
+                /// Enables the internal voltage/sensor
+                /// ADC must be disabled.
```

Comment:
Is it really a must have requirement for the ADC to be disabled for this to work? If so why not try and follow a design pattern based on type states like for the GPIO in order to enforce this at compile time.

Reply:
Great point, the answer is yes the ADC should be disabled. RM0433 page 1028 says:
>Note: The software is allowed to write this bit only when the ADCs are disabled (ADCAL=0, JADSTART=0, ADSTART=0, ADSTP=0, ADDIS=0 and ADEN=0).